### ISSUE TITLE
Creating mbtiles for GCC states fails

### ISSUE THREAD

**Issue by kuwapa** [systemed/tilemaker#332]
Hi, 

I'm continuing my work on creating mbtiles for all countries and while creating the mbtiles for the Gulf countries, tilemaker fails with this error -

```
lua runtime error: Could not find way with id 988493675
terminate called after throwing an instance of 'kaguya::LuaTypeMismatch'
  what():  type mismatch!!
Aborted (core dumped)
```

This is the pbf file - https://download.geofabrik.de/asia/gcc-states-latest.osm.pbf

Command used -
```
tilemaker --input gcc-states-latest.osm.pbf --output gcc.mbtiles --config new_config.json --process new-process-openmaptiles.lua 
```

This is the whole log
```
Layer place (z0-14)
Layer boundary (z0-14)
Layer poi (z12-14)
Layer poi_detail (z14-14) -> poi
Layer housenumber (z14-14)
Layer waterway (z8-14)
Layer waterway_detail (z12-14) -> waterway
Layer transportation (z4-14)
Layer transportation_name (z8-14)
Layer transportation_name_mid (z13-14) -> transportation_name
Layer transportation_name_detail (z14-14) -> transportation_name
Layer building (z14-14)
Layer water (z6-14)
Layer ocean (z0-14) -> water
Layer water_name (z14-14)
Layer water_name_detail (z14-14) -> water_name
Layer aeroway (z11-14)
Layer aerodrome_label (z10-14)
Layer park (z11-14)
Layer landuse (z4-14)
Layer urban_areas (z4-8) -> landuse
Layer landcover (z8-14)
Layer ice_shelf (z0-9) -> landcover
Layer glacier (z2-9) -> landcover
Layer mountain_peak (z11-14)
Bounding box 34.4349, 15.2475, 60.9475, 32.2991
Reading .shp ocean
Reading .shp urban_areas
Unable to open landcover/ne_10m_urban_areas/ne_10m_urban_areas.shp or landcover/ne_10m_urban_areas/ne_10m_urban_areas.SHP.
Reading .shp ice_shelf
Unable to open landcover/ne_10m_antarctic_ice_shelves_polys/ne_10m_antarctic_ice_shelves_polys.shp or landcover/ne_10m_antarctic_ice_shelves_polys/ne_10m_antarctic_ice_shelves_polys.SHP.
Reading .shp glacier
Unable to open landcover/ne_10m_glaciated_areas/ne_10m_glaciated_areas.shp or landcover/ne_10m_glaciated_areas/ne_10m_glaciated_areas.SHP.
Sorting loaded shapes
Reading .pbf gcc-states-latest.osm.pbf
Block 2050/2303 ways 0 relations 0        
Sorting nodes
Block 2299/2303 ways 7905 relations 0        
Sorting ways
lua runtime error: Could not find way with id 988493675
terminate called after throwing an instance of 'kaguya::LuaTypeMismatch'
  what():  type mismatch!!
Aborted (core dumped)
```

What can I do so that this way(988493675) or any other objects for which the process fails are skipped or avoided and the rest of the map is processed? 

Thanks.

**Comment by kuwapa** [comment]
I've modified the process-lua file and config file a bit, but I re-ran the tilemaker with the default files using this command, and the same error still happened

```
tilemaker --input gcc-states-latest.osm.pbf --output gcc.mbtiles --config tilemaker/config.json --process tilemaker/process.lua 
```

**Comment by systemed** [comment]
I think this is probably the same as #324. I'll see if I can work out what's happening.

**Comment by systemed** [comment]
Have a go with #333 - that seems to fix it for me.

**Comment by kuwapa** [comment]
Thanks a ton for this super quick fix! #333 fixed the problem. I tried both gcc-states-latest.osm.pbf and karlsruhe-regbez-latest.osm.pbf mentioned in #324. Mbtiles for both got generated without any hitches and the gets rendered in Maplibre flawlessly!
